### PR TITLE
Dependency management documentation improvements

### PR DESCRIPTION
**Pull request by jjohannes**
Addresses issues listed in #7481 

Rendered version:
- [Excludes](https://builds.gradle.org/repository/download/Gradle_Check_BuildDistributions/27309678:id/distributions/gradle-6.0-all.zip!/gradle-6.0-20191001064544%2B0000/docs/userguide/dependency_downgrade_and_exclude.html)
- [Terminology](https://builds.gradle.org/repository/download/Gradle_Check_BuildDistributions/27309678:id/distributions/gradle-6.0-all.zip!/gradle-6.0-20191001064544%2B0000/docs/userguide/dependency_management_terminology.html)

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/dependency_downgrade_and_exclude.adoc [hunk 1]
```diff
@@ -92,61 +92,80 @@ include::sample[dir="userguide/dependencyManagement/managingTransitiveDependenci
...
+include::sample[dir="userguide/dependencyManagement/managingTransitiveDependencies/excludeForDependency/groovy",files="build.gradle[tags=exclude-transitive-dependencies-3]"]
+include::sample[dir="userguide/dependencyManagement/managingTransitiveDependencies/excludeForDependency/kotlin",files="build.gradle.kts[tags=exclude-transitive-dependencies-3]"]
 ====
 
-You might encounter other use cases that don't quite fit the bill of an exclude rule. For example you want to automatically select a version for a dependency with a specific requested version or you want to select a different group for a requested dependency to react to a relocation. Those use cases are better solved by the link:{groovyDslPath}/org.gradle.api.artifacts.ResolutionStrategy.html[ResolutionStrategy] API. Some of these use cases are covered in <<customizing_dependency_resolution_behavior.adoc#customizing_dependency_resolution_behavior,Customizing Dependency Resolution Behavior>>.
+Historically, excludes are also used as a band aid to fix other issues not supported by some dependency management systems.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/dependency_downgrade_and_exclude.adoc]: excludes were also used

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/dependency_downgrade_and_exclude.adoc [hunk 2]
```diff
@@ -92,61 +92,80 @@ include::sample[dir="userguide/dependencyManagement/managingTransitiveDependenci
...
+Grade however, offers a variaty of features that might be better suited to solve a certain use case.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/dependency_downgrade_and_exclude.adoc]: Gradle however

File: subprojects/samples/src/integTest/groovy/org/gradle/integtests/samples/UserGuideSamplesIntegrationTest.groovy [hunk 3]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package org.gradle.integtests.samples
+    package org.gradle.integtests.samples
```
Comment (ljacomet) [inline subprojects/samples/src/integTest/groovy/org/gradle/integtests/samples/UserGuideSamplesIntegrationTest.groovy]: Not needed white space

File: subprojects/docs/src/docs/userguide/dep-man/dependency_management_terminology.adoc [hunk 4]
```diff
@@ -14,30 +14,45 @@
...
+In such cases, the artifact is a single file.
+Directories are common in the case of inter-project dependencies to avoid the cost of producing the publishable artifact.
+
+[[sub:terminology_capability]]
+== Capability
+
+A capability identifies a feature offered by one or multiple components.
+A capability is identified by coordinates similar to the coordinates used for a <<sub:terminology_module_version,module versions>>.
+By default, each module version offers a capability that matches it's coordinates, for example `com.google:guava:18.0`.
+Capabilities can be used to express that a component provides multiple <<sub:terminology_feature_variant,feature variants>> or that two different components implement the same feature (and can thus not be used together).
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/dependency_management_terminology.adoc]: (and thus cannot be used together)

File: subprojects/docs/src/docs/userguide/dep-man/dependency_management_terminology.adoc [hunk 5]
```diff
@@ -48,30 +63,54 @@ The word "configuration" is an overloaded term and has a different meaning outsi
...
 Dependency constraints can be used to express such requirements for transitive dependencies.
-For more information, see <<dependency_constraints.adoc#sec:adding-constraints-transitive-deps,Adding constraints on transitive dependencies>>.
+For more information, see the sections on <<dependency_constraints.adoc#,upgrading>> and <<dependency_downgrade_and_exclude.adoc#,downgrading>> transitive dependencies.
+
+[[sub:terminology_feature_variant]]
+== Feature Variant
+
+A feature variant is a <<#sub:terminology_variant,variant>> representing a feature of a component that can be individually selected or not.
+A feature variant is identified by one or more <<#sub:terminology_capability,capabilities>>.
+A For more information, see the sections on <<feature_variants.adoc#,modeling feature variants and optional dependencies>>.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/dependency_management_terminology.adoc]: Remove leading A

File: subprojects/docs/src/docs/userguide/dep-man/dependency_management_terminology.adoc [hunk 6]
```diff
@@ -48,30 +63,54 @@ The word "configuration" is an overloaded term and has a different meaning outsi
...
+For convenient consumption, modules can be hosted in a <<#sub:terminology_repository,repository>>.
 
 [[sub:terminology_module_metadata]]
 == Module metadata
 
-Releases of a <<#sub:terminology_module,module>> can provide metadata. Metadata is the data that describes the module in more detail e.g. the coordinates for locating it in a repository, information about the project or required <<#sub:terminology_transitive_dependency,transitive dependencies>>. In Maven the metadata file is called `.pom`, in Ivy it is called `ivy.xml`.
+Releases of a <<#sub:terminology_module,module>> provide metadata.
+Metadata is the data that describes the module in more detail e.g. information about the location of artifacts or required <<#sub:terminology_transitive_dependency,transitive dependencies>>.
+Gradle offers it's own metadata format called link:https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/design/gradle-module-metadata-latest-specification.md[Gradle Module Metadata] (`.module` file) but also supports Maven (`.pom`) and Ivy (`ivy.xml`) metadata.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/dependency_management_terminology.adoc]: Gradle offers its own metadata

**Comment by jjohannes** [comment]
Thanks @ljacomet! I fixed the typos/formatting issues.

**Pull request merged**